Source/Objects/Custom Hardware/SNO+/CAEN/SNOCaenModel.m

```
@@ -1252,6 +1274,65 @@ - (void) encodeWithCoder:(NSCoder*) anEncoder
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedLong:[self customSize]] forKey:[self getStandardRunKeyForField:@"customSize" ]];
+        [CAENStateDict setObject:[NSNumber numberWithBool:[self isCustomSize]] forKey:[self getStandardRunKeyForField:@"isCustomSize" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedShort:[self acquisitionMode]] forKey:[self getStandardRunKeyForField:@"acquisitionMode" ]];
+        [CAENStateDict setObject:[NSNumber numberWithBool:[self countAllTriggers]] forKey:[self getStandardRunKeyForField:@"countAllTriggers" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedLong:[self triggerSourceMask]] forKey:[self getStandardRunKeyForField:@"triggerSourceMask" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedShort:[self coincidenceLevel]] forKey:[self getStandardRunKeyForField:@"coincidenceLevel" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedLong:[self triggerOutMask]] forKey:[self getStandardRunKeyForField:@"triggerOutMask" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedLong:[self postTriggerSetting]] forKey:[self getStandardRunKeyForField:@"postTriggerSetting" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedLong:[self frontPanelControlMask]] forKey:[self getStandardRunKeyForField:@"frontPanelControlMask" ]];
+        [CAENStateDict setObject:[NSNumber numberWithUnsignedShort:[self enabledMask]] forKey:[self getStandardRunKeyForField:@"enabledMask" ]];
+        for (int idac=0; idac<kNumCaenChannelDacs; ++idac) {
+            [CAENStateDict setObject:[NSNumber numberWithUnsignedShort:[self dac:idac]] forKey:[NSString stringWithFormat:@"%@_%i",[self getStandardRunKeyForField:@"dac"],idac]];
+        }
+    } @catch(...){
+        NSLogColor([NSColor redColor], @"CAEN: settings couldn't not be saved \n");
```

If you're going to catch the exception here, you need to relay that info back to the caller so it doesn't save a dictionary with invalid values. There are two ways to do this:

1. don't catch the exception. Then you need to make sure any upstream code has exception handlers.
2. Pass back NULL instead of an NSDictionary pointer, and then make sure all calls to this function check the return value.

I think the best way depends on how often something like this will happen. Since these are just getting objects from the model it's very unlikely to fail, and so I would just let the exception bubble up and put try catch blocks around the code which gets the current state.